packages/mask/src/plugins/Collectible/pipes/index.ts

```diff
@@ -79,9 +87,26 @@ export function resolveTraitLinkOnOpenSea(chainId: ChainId, slug: string, search
     return `https://opensea.io/assets/${slug}?search[stringTraits][0][name]=${search}&search[stringTraits][0][values][0]=${value}`
 }
 
-export function resolveAssetLinkOnOpenSea(chainId: ChainId, address: string, id: string) {
-    return urlcat(
-        resolveLinkOnOpenSea(chainId === ChainId.Mainnet ? ChainId.Mainnet : ChainId.Rinkeby),
-        `/assets/${address}/${id}`,
-    )
+export function resolveAssetLinkOnCurrentProvider(
+    chainId: ChainId,
+    address: string,
+    id: string,
+    provider: NonFungibleAssetProvider,
+) {
+    switch (provider) {
+        case NonFungibleAssetProvider.OPENSEA:
+            return urlcat(
+                resolveLinkOnOpenSea(chainId === ChainId.Mainnet ? ChainId.Mainnet : ChainId.Rinkeby),
```

Comment:
plz extract chain id computed result as variable to this function top-level

```suggestion
                resolveLinkOnOpenSea(computedChainId),
```